share/availability/full/availability.go

```diff
@@ -23,23 +24,44 @@ var log = logging.Logger("share/full")
 type ShareAvailability struct {
 	store  *store.Store
 	getter shwap.Getter
+
+	storageWindow time.Duration
+	archival      bool
 }
 
 // NewShareAvailability creates a new full ShareAvailability.
 func NewShareAvailability(
 	store *store.Store,
 	getter shwap.Getter,
+	opts ...Option,
 ) *ShareAvailability {
+	p := defaultParams()
+	for _, opt := range opts {
+		opt(p)
+	}
+
 	return &ShareAvailability{
-		store:  store,
-		getter: getter,
+		store:         store,
+		getter:        getter,
+		storageWindow: p.storageWindow,
+		archival:      p.archival,
 	}
 }
 
 // SharesAvailable reconstructs the data committed to the given Root by requesting
 // enough Shares from the network.
 func (fa *ShareAvailability) SharesAvailable(ctx context.Context, header *header.ExtendedHeader) error {
 	dah := header.DAH
+
+	if !fa.archival {
+		// do not sync blocks outside of sampling window if not archival
+		if !availability.IsWithinWindow(header.Time(), fa.storageWindow) {
+			log.Debugw("skipping availability check for block outside sampling"+
+				" window", "height", header.Height(), "data hash", dah.String())
```

Comment:
Just curious why to split `window` word?

Reply:
Unintentional, it's just string